### PR TITLE
fix: parallelize registry builds

### PR DESCRIPTION
**Pull request by marshacb**
* Parallelizes querys to nodes to improve registry build times
* Skips processing certain RPC endpoints throwing abnormal closures/errors which caused registry builds to fail
* Removes `XcmV3MultiLocation` type construction and replaces with custom XCM MultiLocation type

closes: [#81](https://github.com/paritytech/asset-transfer-api-registry/issues/81)

### CODE REVIEW

File: src/createRegistry.ts [hunk 1]
```diff
@@ -46,6 +49,24 @@ const WS_DISCONNECT_TIMEOUT_SECONDS = 3;
...
 	'https://cdn.jsdelivr.net/gh/colorfulnotion/xcm-global-registry/metadata/xcmgar.json';
 
+/**
+ * Determines when endpoint processing should be skipped.
+ *
+ * @param endpoint
+ */
+const skipProcessingEndpoint = (endpoint: string): boolean => {
+	if (
+		endpoint.includes('onfinality') ||
```
Comment (TarikGul) [inline src/createRegistry.ts:59]: I think we should have a rpc blacklist array for these specific endpoints. So Something like:

`endpoint.includes('onfinality') || blackLists.includes(endpoint)`

Comment (marshacb) [reply src/createRegistry.ts]: makes sense to me. Added an array for these values instead

**Review by TarikGul — APPROVED** [review]
LGTM. 

I think this is a great first step to parallelizing the calls. But we should continue to increase the performance by isolating the connections into parallel connections. So currently we just add each chains process as a single clump into a Promise.all, but if we want to make it even faster we can take each rpc connection under a single chain and run them all at the same time.

Here's how we do it in sidecar for `/blocks?range=`: https://github.com/paritytech/substrate-api-sidecar/blob/master/src/util/PromiseQueue.ts

We can also use generators in JS to achieve this but the above Promise queue is a little faster. We can also set a max size so we don't run to many connection at parallel.

**Comment by TarikGul** [comment]
Let's merge this in as its a big upgrade, then we can focus on increasing performance even more. It might take a bit of refactoring to get each rpc connection isolated into a Promise Queue.

**Pull request merged**